### ISSUE TITLE
Error: frame header bad format when splitting Audiobooks

### ISSUE THREAD

**Issue by mqus** [gbouthenot/mp3splitter-js#1]
I found your script while looking for ways to split the audiobook mp3s generated by openaudible
(https://github.com/openaudible/openaudible/issues/43)
But I tried out multiple audiobooks and they all failed with the similar error:
```
$ node mp3splitter.js dx/Der\ Kleine\ Prinz.mp3 
Uint8Array [ 84, 88, 88, 88, 0, 0, 1, 193, 0, 0 ]
/tmp/did/mp3splitter-js/mp3splitter.js:678
                throw new Error('frame header bad format')
                ^

Error: frame header bad format
    at Mp3splitter.go (/tmp/did/mp3splitter-js/mp3splitter.js:678:23)
    at Cliapp.go (/tmp/did/mp3splitter-js/mp3splitter.js:35:17)
    at Object.<anonymous> (/tmp/did/mp3splitter-js/mp3splitter.js:750:8)
    at Module._compile (internal/modules/cjs/loader.js:701:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:712:10)
    at Module.load (internal/modules/cjs/loader.js:600:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:539:12)
    at Function.Module._load (internal/modules/cjs/loader.js:531:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:754:12)
    at startup (internal/bootstrap/node.js:283:19)
```
I didn't change the output of openaudible at all and the mp3s can be played on all the players I tried. The ID3 Tags also didn't seem to have an error in the players (I didn't look at them closely)

**Comment by artempleton** [comment]
I encountered the same error. I'm using nodejs 12.3.1 installed via Homebrew on macOS 10.14.5. 

```
/.../mp3/mp3splitter.js:688
                throw new Error('frame header bad format')
                ^

Error: frame header bad format
    at Mp3splitter.go (/.../mp3/mp3splitter.js:688:23)
    at Cliapp.go (/.../mp3/mp3splitter.js:35:17)
    at Object.<anonymous> (/.../mp3/mp3splitter.js:760:8)
    at Module._compile (internal/modules/cjs/loader.js:774:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:785:10)
    at Module.load (internal/modules/cjs/loader.js:641:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:837:10)
    at internal/main/run_main_module.js:17:11
```

**Comment by gbouthenot** [comment]
Thank you for your report !
I will investigate these errors.
Seem they are not related (the report from @artempleton does not show the buffer, don't know why).
Report from @mqus is related to a (badly ?) formatted size, strange...

Is there any possibility you send me the start of the file ? (the first 600 Ko or so ?) please put it somewhere and send me the link by mail (get my email at https://github.com/gbouthenot)
Thank you.

**Comment by gbouthenot** [comment]
According to the id3 specs (http://id3.org/id3v2.4.0-structure), the size is a syncsafe integer, so the numbers in [ 0, 0, 1, 193 ], MUST NOT be >=128. The tag is not standard compliant. Can't tell more without the file. Maybe skipping the validation is the way to do ?
```
     Frame ID      $xx xx xx xx  (four characters)
     Size      4 * %0xxxxxxx
     Flags         $xx xx

   In some parts of the tag it is inconvenient to use the
   unsychronisation scheme because the size of unsynchronised data is
   not known in advance, which is particularly problematic with size
   descriptors. The solution in ID3v2 is to use synchsafe integers, in
   which there can never be any false synchs. Synchsafe integers are
   integers that keep its highest bit (bit 7) zeroed, making seven bits
   out of eight available. Thus a 32 bit synchsafe integer can store 28
   bits of information.
   
   Example:
     255 (%11111111) encoded as a 16 bit synchsafe integer is 383
     (%00000001 01111111).
```

@mqus : can you try commenting ```return false```, in the ```readSyncSafeInt32``` method ?

@artempleton: Maybe you did not include the ```Uint8Array [ ... ]``` line just before the error ? Can you ?

Both: send me the start of the file as stated above, so I can see what is wrong.

**Comment by mqus** [comment]
I sent you the first MiByte of the audiobook. I'll try your fix now

**Comment by mqus** [comment]
I probably found the error: my files have version 2.3 of the standard. In this version, the frame headers are not "syncsafe" because 2.3 uses an "unsynchronization scheme" everywhere except the ID3 header (http://id3.org/id3v2.3.0#The_unsynchronisation_scheme). This means that the header size bytes can be any value and not only <128  (http://id3.org/id3v2.3.0#ID3v2_header).

After uncommenting the `return false`  I got another error: It jumped into a tag and misinterpreted it as a header because the read frame size was wrong (obviously, in hindsight).

**Comment by gbouthenot** [comment]
You were right.
ID3v2.3 was not handled correcty. The documentation of id3v2.4 does not say anything about previous versions.
I pushed a commit to handle ID3v2.3 and ID3v2.4 tag. Can you tell me if it fixes your problem ?

**Comment by artempleton** [comment]
@gbouthenot, sorry I didn't provide all the info you needed yesterday. However, I can happily report that the recently committed code handled the split just fine. Worked like a charm. Many thanks!

**Comment by gbouthenot** [comment]
Thanks both of you. I consider this issue closed.
Don't hesitate to re-open if you still have problems.

Fixed by 23bfd33db10ca9242f46ea1ad730af242bfad49d

**Comment by mqus** [comment]
It's fixed for me, thanks!
